app/lib/common/utils/utils.dart

```diff
@@ -70,8 +71,27 @@ extension RefDebounceExtension on Ref {
 
 const largeScreenBreakPoint = 770;
 
-bool isLargeScreen(BuildContext context) {
-  return MediaQuery.of(context).size.width >= largeScreenBreakPoint;
+extension ActerContextUtils on BuildContext {
+  bool get isLargeScreen =>
+      MediaQuery.of(this).size.width >= largeScreenBreakPoint;
+
+  NavigatorState closeDialog({Routes? orGo, bool ignorePopFailure = false}) {
+    final navigator = Navigator.of(this);
+    try {
+      if (orGo != null && !navigator.canPop()) {
+        navigator.pushReplacementNamed(orGo.name);
+      } else {
+        navigator.pop();
```

Comment:
I honestly don't really understand why and when this would be needed. Are we sure _it is needed_ and not just the common way replacement of a broken `context.pop`?

But the `closeDialog` doesn't replace any `Navigator.of(context, rootNavigator: true).pop()` at this point either way, so ...

Reply:
Noticed how even the cookbook example does not use `Navigator.of(context, rootNavigator: true).pop()` when inside the `showDialog` (despite that using `useRootNavigator:true` by default) - https://docs.flutter.dev/cookbook/effects/nested-nav . IF the right `context` is used (the one provided in the build function for the `showDialog`) this should work just fine.

Reply:
> I honestly don't really understand why and when this would be needed. Are we sure it is needed and not just the common way replacement of a broken context.pop?

Don't exactly remember but in some of the cases while closing dialogs with `context.pop` , it actually close screen open behind it and dialog keeps remain open. So in those cases `rootNavigator` help in closing dialog first.

Let me share one of the screen recoding for reference.

https://github.com/user-attachments/assets/a3131716-0716-4de2-98d4-2eebc1b088b2

In Above video, on click of close button it actually closing up the dialog first, but not exactly remember many time I see the issue like dialog keeps remain open on click of close button also where context.pop() written.

> But the closeDialog doesn't replace any Navigator.of(context, rootNavigator: true).pop() at this point either way, so ...

👍

Reply:
@kumarpalsinh25 but that isn't due to the `context.pop` but instead that we are putting the dialog onto the wrong navigator. the back-button pops on the latest, but the showDialogs are plugged to the root.

This fixes that: https://github.com/acterglobal/a3/pull/1997#issue-2430426341